Documentation/Porting.md

```diff
@@ -0,0 +1,251 @@
+# Porting to new platforms
+
+<!--
+This source file is part of the Swift.org open source project
+
+Copyright (c) 2024 Apple Inc. and the Swift project authors
+Licensed under Apache License v2.0 with Runtime Library Exception
+
+See https://swift.org/LICENSE.txt for license information
+See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+
+When the Swift toolchain is ported to a new platform, it is necessary to port
+Swift Testing as well. This document contains various information, trivia, and
+deep wisdoms about porting Swift Testing.
+
+> [!NOTE]
+> This document uses Classic Mac OS ("Classic") as an example target platform.
+> In this hypothetical scenario, we assume that the Swift compiler identifies
+> Classic with `os(Classic)` and that the C++ compiler identifies it with
+> `defined(macintosh)`. Other platforms would be identified differently.
+
+## Getting started
+
+Before you start the porting process, make sure you are very familiar with Swift
+and C++ as well as the C standard library and platform SDK for your target
+platform.
+
+Your first task when porting Swift Testing is ensuring that it builds.
+
+We've made an effort to ensure that as much of our code as possible is
+platform-agnostic. When building the toolchain for a new platform, you will
+hopefully find that Swift Testing builds out-of-the-box with few, if any,
+errors.
+
+> [!NOTE]
+> Swift Testing relies on the Swift [standard library](https://github.com/swiftlang/swift),
+> Swift macros (including the [swift-syntax](https://github.com/swiftlang/swift-syntax) package),
+> and [Foundation](https://github.com/apple/swift-foundation). These components
+> must build and (minimally) function before you will be able to successfully
+> build Swift Testing regardless of which platform you are porting to.
+
+### Swift or C++?
```

Comment:
Perhaps reorder this section later in the document? Reading through it as a step-by-step, this paragraph feels a little bit out of place. What changes would I even be making, where the choice of programming language is applicable?

Reply:
If I'm reading along, I want to know "you should be using Swift" before I get into the technical details, I think? It would read oddly to be two sections in and see "oh by the way don't use C++ unless you absolutely have to." I've already started writing code by then!

I think.

I dunno. Where do you want to move it to?

Reply:
Going to merge this as-is, but we can revise.